tesseract-server/src/handlers/logic_layer/aggregate.rs

```diff
@@ -335,6 +341,35 @@ pub fn logic_layer_aggregation(
 }
 
 
+// Genrates the source data/ annotaion of the cube for which the query is executed
+fn generate_source_data(cube: &Cube) -> Option<SourceMetadata> {
+    let cube_name = &cube.name;
+    let mut measures = Vec::new();
+    // let mut annotations = HashMap::new();
+    for measure in cube.measures.iter() {
+        measures.push(measure.name.clone());
+    }
+    let annotations = match cube.annotations.clone(){
+        Some(annotations) => {
+            let mut anotate_hashmap = HashMap::new();
+            for annotation in annotations.iter(){
+                anotate_hashmap.entry(annotation.name.to_string()).or_insert(Vec::new()).push(annotation.text.to_string());
+            }
+            Some(anotate_hashmap)
+        },
+        None => None
+    };
+    // println!("{:?}", cube_name);
+    // println!("{:?}", measures);
+    // println!("{:?}", annotations);
+    Some(SourceMetadata {
+        name: cube_name.clone(),
+        measures: measures.clone(),
```

Comment:
I don't think we need to include the measure list here.

Reply:
never mind, I see in Dave's examples he did include measures so we can keep this